gpm_api/tests/io/test_checks.py

```diff
@@ -0,0 +1,478 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Jul 17 15:41:14 2023
+
+@author: ghiggi
+"""
+
+import pytest
+import datetime
+import numpy as np
+from gpm_api.io import checks
+from gpm_api.io.products import available_products, available_scan_modes
+
+
+def test_is_not_empty() -> None:
+    ''' Test is_not_empty() '''
+
+    # Test a non empty object
+    res = checks.is_not_empty([1, 2, 3])
+    assert res is True, "Function returned False, expected True"
+
+    # Test an empty object
+    res = checks.is_not_empty([])
+    assert res is False, "Function returned True, expected False"
+
+
+def test_is_empty() -> None:
+    ''' Test is_empty() '''
+
+    # Test a non empty object
+    res = checks.is_empty([1, 2, 3])
+    assert res is False, "Function returned True, expected False"
+
+    # Test an empty object
+    res = checks.is_empty([])
+    assert res is True, "Function returned False, expected True"
+
+
+def test_check_base_dir() -> None:
+    """ Check path constructor for base_dir """
+
+    # Check leading slash is removed
+    res = checks.check_base_dir('/home/user/gpm/')
```

Comment:
Check also for `/home/user/gpm` ... the current test checks that the last `/` is removed.
Can we also test (and maybe adapt the check function) for with windows paths ?
And check/test that the path is correct for the system in use?

Reply:
I've added the firs test. I'm not quite sure about Windows, maybe @sphamba if you could help here. If I run Windows tests locally (on Linux) they won't work I think, do we need to call `if platform.system() == "Windows"` ?

Reply:
@evanjt define the test filepath assembling it with `os.path.join`
And also check that if on linux does not contain \ and on window / ...
`os.path.sep` to derive the correct slash ... and `pathlib` I guess also has some utility for that

Reply:
@evanjt I agree on having a `os.name == "nt"` or `platform.system() == "Windows"` check in the test, then constructing the tested path using `os.path.join` and appending `os.path.sep`. Also change the [source](https://github.com/EPFL-ENAC/gpm_api/blob/1e791373fc07db16417f9a673e79fc143f93682d/gpm_api/io/checks.py#L35) with `os.path.sep`